tests/uefistored/conftest.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+from contextlib import contextmanager
+from collections import namedtuple
+import ntpath
+import shlex
+
+from lib.common import is_uuid, wait_for
+from lib.efi import (
+    Certs,
+    EFIAuth,
+    EFI_AT_ATTRS,
+    EFI_GUIDS,
+)
+
+VM_SECURE_BOOT_FAILED = 'VM_SECURE_BOOT_FAILED'
+
+@pytest.fixture
+def sb_vm_unsigned(running_vm):
+    vm = running_vm
+
+    print('> rm uefistored guest certs')
+    vm.host.ssh(['varstore-sb-state', vm.uuid, 'setup'])
+
+    print('> rm uefistored host certs in /usr/share/varstored/')
+    vm.host.ssh(['rm', '/usr/share/varstored/{KEK.auth,db.auth,dbx.auth}'],
+                check=False)
+
+    print('> clear guest SB messages')
+    vm.rm_messages(VM_SECURE_BOOT_FAILED)
+
+    snapshot = vm.snapshot()
```

Comment:
I would understand a snapshot before any modifications to the VM, but not after the above clean-up. Plus, the snapshot will only ever get reverted when all tests that depend on the fixture have run, so in case you were counting on it to restore the VM state in-between tests, this won't work.

Reply:
The `snapshot()` actually does revert between tests, as this is function-scoped.  I'm just using it here to revert to a cleared slate of UEFI variables between tests.

The output looks like:

```
> clear guest SB messages
Snapshot VM
> Setting PK
Shutdown VM
> VM is halted
Start VM
Wait for VM running
Wait for VM IP
VM IP: 192.168.0.14
Wait for management agent up
PASSEDRevert snapshot

tests/uefistored/test_secure_boot.py::test_boot_fails_when_db_set_and_images_unsigned[192.168.0.17-alpine-linux-x86_64-uefi] > rm uefistored guest certs
> rm uefistored host certs in /usr/share/varstored/
> clear guest SB messages
Snapshot VM
> Setting PK
> Setting KEK
> Setting db
Start VM
Wait for message VM_SECURE_BOOT_FAILED
PASSEDRevert snapshot
```

Reply:
The only alternative I could see was to make a function-scoped `imported_vm` fixture, which would be far slower.

Reply:
Oh, I missed the fact that it's function-scoped. All good then.